### PR TITLE
HPCC-31258 Initial File structure and page layout creation

### PR DESCRIPTION
**Pull request by kunalaswani**
This is to create the initial design of Sasha in Cloud. Feature availability to come as it is created.

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browsers
  - [x] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by jeclrsg — COMMENTED** [review]
@kunalaswani There are a lot of lint errors getting annotated by GH Actions

File: esp/src/src-react/routes.tsx [hunk 1]
```diff
@@ -390,6 +390,11 @@ export const routes: RoutesEx = [
                     },
                 ]
             },
+            {
+                path: "/sasha", action: (ctx, params) => import("./components/Sasha").then(_ => {
```
Comment (jeclrsg) [inline esp/src/src-react/routes.tsx:394]: Should look more like the routing changes you'd made for the Dali admin, with child routes based on which "Tab" of the UI was selected?

File: esp/src/src-react/components/Sasha.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
+import React, { useState } from "react";
+
```
Comment (jeclrsg) [inline esp/src/src-react/components/Sasha.tsx:2]: This component should probably be using the FluentUI components for any HTML it returns.

**Review by jeclrsg — CHANGES_REQUESTED** [review]
I would think this should be more similar to your Dali Admin PR(s), both with respect to the changes to routes.tsx and the Sasha.tsx component, ie using FluentUI components.

**Review by jeclrsg — APPROVED** [review]

File: esp/src/src-react/components/Sasha.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  return (
+    <div>
+      <form onSubmit={handleSubmit}>
+        <Dropdown
+          placeholder="Select an option"
+          selectedKey={selectedOption}
+          onChange={handleOptionChange}
+          options={[
+            { key: "", text: "Select an option" },
```
Comment (ghalliday) [inline esp/src/src-react/components/Sasha.tsx:63]: Should these display strings be i18n?

**Review by ghalliday — COMMENTED** [review]
See comment about embedded text

File: esp/src/src-react/components/Sasha.tsx (hunk 3 above)
Comment (kunalaswani) [reply esp/src/src-react/components/Sasha.tsx]: @ghalliday corrected

**Comment by ghalliday** [comment]
@kunalaswani if you are making changes to a PR, please push them as separate commits, which can than be squashed later - otherwise it is hard to spot the changes.  Also, you need to re-request reviews.

**Review by ghalliday — APPROVED** [review]

File: esp/src/src-react/components/Sasha.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        console.log("Invalid option selected");
+    }
+    // Reset form
+    setSelectedOption("");
+    setWuid("");
+    setResult("");
+  };
+
+  // Conditional rendering for default value
+  const defaultValue = result ? null : <div>No data available</div>;
```
Comment (jeclrsg) [inline esp/src/src-react/components/Sasha.tsx:54]: "No data available" should be in translations

File: esp/src/src-react/components/Sasha.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+  return (
+    <div>
+      <form onSubmit={handleSubmit}>
+        <Dropdown
+          placeholder="Select an option"
```
Comment (jeclrsg) [inline esp/src/src-react/components/Sasha.tsx:60]: should be nlsHPCC.SelectAnOption

File: esp/src/src-react/components/Sasha.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+            { key: "archiveDFUWorkUnit", text: nlsHPCC.ArchiveDFUWorkunit },
+            { key: "backupECLWorkUnit", text: nlsHPCC.BackupECLWorkunit },
+            { key: "backupDFUWorkUnit", text: nlsHPCC.BackupDFUWorkunit }
+          ]}
+          styles={{ dropdown: { width: 400 } }}
+        />
+        {["restoreECLWorkUnit", "restoreDFUWorkUnit", "archiveECLWorkUnit", "archiveDFUWorkUnit", "backupECLWorkUnit", "backupDFUWorkUnit"].includes(selectedOption) && (
+          <div>
+            <TextField
+              label="Work Unit ID:"
```
Comment (jeclrsg) [inline esp/src/src-react/components/Sasha.tsx:79]: There's an nlsHPCC.WUID, or this one should be added

**Review by jeclrsg — CHANGES_REQUESTED** [review]
@kunalaswani just noticed a couple other small things, other strings that should be using the translations file

**Review by jeclrsg — APPROVED** [review]

**Pull request merged**